fcrepo4/fcrepo4
# Removed constrainedBy link from successful GET responses.

**JIRA Ticket**: https://jira.duraspace.org/browse/FCREPO-2894

# What does this Pull Request do?
Removed constrainedBy link from successful GET responses.

# What's new?
Removed constrainedBy link from successful GET responses, to bring the implementation in line with the spec (which doesn't require constrainedBy on successful GET requests). Also removed tests that were checking for that link.

# How should this be tested?

A regular `mvn clean install` should do it. Also can check using `curl` that there are no `constrainedBy` link headers:

```
curl -u fedoraAdmin:fedoraAdmin http://localhost:8080/rest -I -L
curl -u fedoraAdmin:fedoraAdmin http://localhost:8080/rest/container -X PUT
curl -u fedoraAdmin:fedoraAdmin http://localhost:8080/rest/container -I
curl -u fedoraAdmin:fedoraAdmin http://localhost:8080/rest/container -i
```

# Additional Notes:
None

# Interested parties
@awoods, @fcrepo4/committers
